src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ((a <= 0 || b <= 0 || c <= 0)
+      || (a + b + c - Math.max(a, b, c)
+       <= Math.max(a, b, c))) {
+      throw new Error('Invalid triangle sides');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const { a, b, c } = this;
+    const sP: number = (a + b + c) / 2;
 
+    return Math.trunc(Math.sqrt(sP * (sP - a) * (sP - b) * (sP - c))
+      * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid radius');
+    }
+  }
+
+  getArea(): number {
+    return Math.trunc(Math.PI * this.radius * this.radius * 100) / 100;
```

Comment:
Remember about DRY. You have the same logic in other classes. Create a function for this `Math.trunc(... * 100) / 100;`